### PR TITLE
Allow to disable TLS with --no-tls

### PR DESCRIPTION
**Pull request by Martoon-00**

### CODE REVIEW

File: wallet-new/server/Cardano/Wallet/Server/CLI.hs [hunk 1]
```diff
@@ -1,22 +1,23 @@
+{-# LANGUAGE ApplicativeDo #-}
```
Comment (adinapoli-iohk) [inline wallet-new/server/Cardano/Wallet/Server/CLI.hs]: Sorry to be a pain in the ass, but any chance we can *not* use `ApplicativeDo`? It feels a little weird that we are using a package like `optparse-applicative` abusing the `do` notation. I know we are doing this virtually everywhere in cardano, but I thought that for new code we could try something different, if possible 😉

File: wallet-new/server/Cardano/Wallet/Server/Plugins.hs [hunk 2]
```diff
@@ -79,11 +76,11 @@ walletBackend :: (HasConfigurations, HasCompileInfo)
               -> Plugin WalletWebMode
 walletBackend WalletBackendParams {..} =
     first one $ worker walletServerOuts $ \sendActions -> do
-      walletServeImpl
+      trace @Text ("Wallet tls: " <> show walletTLSParams) $ walletServeImpl
```
[the remark this answers is not in the document]
Comment (Martoon-00) [reply wallet-new/server/Cardano/Wallet/Server/Plugins.hs]: Oh right, forgot about it

File: wallet-new/server/Cardano/Wallet/Server/CLI.hs (hunk 1 above)
Comment (Martoon-00) [reply wallet-new/server/Cardano/Wallet/Server/CLI.hs]: When I've read this ^, I realized what was wrong :))
Arguments were passed in wrong order to `TlsParams` constructor

**Comment by Martoon-00** [comment]
Fixed

**Pull request merged**